src/toil_lib/urls.py

```diff
@@ -37,77 +42,107 @@ def download_url(url, work_dir='.', name=None, s3_key_path=None, cghub_key_path=
 def download_url_job(job, url, name=None, s3_key_path=None, cghub_key_path=None):
     """Job version of `download_url`"""
     work_dir = job.fileStore.getLocalTempDir()
-    fpath = download_url(url, work_dir=work_dir, name=name,
+    fpath = download_url(job=job, url=url, work_dir=work_dir, name=name,
                          s3_key_path=s3_key_path, cghub_key_path=cghub_key_path)
     return job.fileStore.writeGlobalFile(fpath)
 
 
-def _download_with_genetorrent(url, file_path, cghub_key_path):
+def _download_with_genetorrent(job, url, file_path, cghub_key_path):
     parsed_url = urlparse(url)
     analysis_id = parsed_url.path[1:]
     assert parsed_url.scheme == 'gnos', 'Improper format. gnos://cghub/ID. User supplied: {}'.format(parsed_url)
     work_dir = os.path.dirname(file_path)
     folder_path = os.path.join(work_dir, os.path.basename(analysis_id))
     parameters = ['-vv', '-c', cghub_key_path, '-d', analysis_id]
-    docker_call(tool='quay.io/ucsc_cgl/genetorrent:3.8.7--9911761265b6f08bc3ef09f53af05f56848d805b',
+    docker_call(job=job, tool='quay.io/ucsc_cgl/genetorrent:3.8.7--9911761265b6f08bc3ef09f53af05f56848d805b',
                 work_dir=work_dir, parameters=parameters)
     sample = glob.glob(os.path.join(folder_path, '*tar*'))
     assert len(sample) == 1, 'More than one sample tar in CGHub download: {}'.format(analysis_id)
 
 
-def s3am_upload(fpath, s3_dir, num_cores=1, s3_key_path=None):
+def s3am_upload(job, fpath, s3_dir, num_cores=1, s3_key_path=None):
     """
     Uploads a file to s3 via S3AM
     S3AM binary must be on the PATH to use this function
     For SSE-C encryption: provide a path to a 32-byte file
 
+    :param toil.job.Job job: Toil job that is calling this function
     :param str fpath: Path to file to upload
     :param str s3_dir: Ouptut S3 path. Format: s3://bucket/[directory]
     :param int num_cores: Number of cores to use for up/download with S3AM
     :param str s3_key_path: (OPTIONAL) Path to 32-byte key to be used for SSE-C encryption
     """
-    require(s3_dir.startswith('s3://'), 'Format of s3_dir (s3://) is incorrect: {}'.format(s3_dir))
+    require(s3_dir.startswith('s3://'), 'Format of s3_dir (s3://) is incorrect: %s' % s3_dir)
```

Comment:
Looks like it's the wrong require. It should be the one from BPL. You may have to bump BPL to the latest release. That's safe because BPL releases are strictly backwards-compatible.